c-layer/lib.cxx

```diff
@@ -86,35 +92,51 @@ static zypp::ZYpp::Ptr zypp_ptr() {
   return NULL;
 }
 
-// TODO: split init target into set of repo manager, initialize target and load target and merge it in rust
-void init_target(const char *root, struct Status *status, ProgressCallback progress, void *user_data) noexcept {
+// TODO: split init target into set of repo manager, initialize target and load
+// target and merge it in rust
+struct Zypp *init_target(const char *root, struct Status *status,
+                         ProgressCallback progress, void *user_data) noexcept {
+  if (the_zypp.zypp_pointer != NULL) {
+    status->state = status->STATE_FAILED;
+    status->error = strdup("Cannot have two init_target concurrently, "
+                           "libzypp not ready for this.");
```

Comment:
:+1: Maybe I would just replace `libzypp not ready for this` with `call free_zypp on the initial one first`.